main/src/ca/uwaterloo/flix/language/errors/NameError.scala

```diff
@@ -234,4 +234,17 @@ object NameError {
     }
   }
 
+  case class MismatchedTypeParamKinds(name: String, loc1: SourceLocation, kind1: Kind, loc2: SourceLocation, kind2: Kind) extends NameError {
+    def summary: String = "Mismatched type parameter kinds."
+    def message: VirtualTerminal = {
+      val vt = new VirtualTerminal
+      vt << Line(kind, source.format) << NewLine
+      vt << ">> Mismatched kinds for type parameter '" << Red(name) << "'." << NewLine
+      vt << NewLine
+      vt << Code(loc1, s"Kind: $kind1") << NewLine // TODO might need FormatKind
+      vt << NewLine
+      vt << Code(loc2, s"Kind: $kind2") << NewLine
```

Comment:
Do you think there is some tip we can write to the programmer? I tried to come up with something:

"You are using a type variable with different kinds. For example:

- ...
"

Actually, how many ways can it even go wrong? Should be just a handful? Maybe we can enumerate them?

Reply:
I've added some possible fixes. I think the only scenarios are unlabelled schema/record and just mislabelling/typo.